driver/src/price_finding/optimization_price_finder.rs

```diff
@@ -239,7 +275,7 @@ impl PriceFinding for OptimisationPriceFinder {
         state: &models::AccountState,
     ) -> Result<models::Solution, PriceFindingError> {
         let input = solver_input::Input {
-            tokens: serialize_tokens(&orders),
+            tokens: serialize_tokens(&orders, self.token_data.clone()),
```

Comment:
Since we only read the `token_data` for prices here, we can probably pass this in by reference instead of cloning.